### ISSUE TITLE
E1019 false positive on transfer family policy parameters

### ISSUE THREAD

**Issue by whoDoneItAgain** [aws-cloudformation/cfn-lint#2477]
### CloudFormation Lint Version

0.71.0

### What operating system are you using?

windows

### Describe the bug

E1019 falsely trips on the 3 transfer policy parameters.

https://docs.aws.amazon.com/transfer/latest/userguide/session-policy.html

### Expected behavior

E1019 doesn't trip on these parameters

### Reproduction template

```
AWSTemplateFormatVersion: 2010-09-09

Resources:

# Transfer User
  TransferUser:
    Type: AWS::Transfer::User
    Properties:
      HomeDirectoryMappings:
        - Entry: /
          Target: /bucket/
      HomeDirectoryType: LOGICAL
      Policy: !Sub |
        {
          "Version": "2012-10-17",
          "Statement": [
            {
              "Sid": "AllowListingOfUserFolder",
              "Action": [
                "s3:ListBucket"
              ],
              "Effect": "Allow",
              "Resource": [
                "arn:${AWS::Partition}:s3:::${transfer:HomeBucket}"
              ],
              "Condition": {
                "StringLike": {
                  "s3:prefix": [
                    "${transfer:HomeFolder}/*",
                    "${transfer:HomeFolder}"
                  ]
                }
              }
            },
            {
              "Sid": "HomeDirObjectAccess",
              "Effect": "Allow",
              "Action": [
                "s3:PutObject",
                "s3:GetObject",
                "s3:DeleteObject",
                "s3:GetObjectVersion"
              ],
              "Resource": "arn:${AWS::Partition}:s3:::${transfer:HomeDirectory}*"
            }
          ]
        }
      Role: replace-me
      ServerId: server-id
      UserName: username
```

**Comment by kddejong** [comment]
Can you provide some more details. This is what I get when trying your template:
`Template format error: Unresolved resource dependencies [transfer:HomeFolder, transfer:HomeBucket, transfer:HomeDirectory] in the Resources block of the template`  You sure you aren't suppose to be escaping those values?

**Comment by whoDoneItAgain** [comment]
you're right. i missed that when adding the policy as described in the link into the sub. sorry about that.
